### PR TITLE
CIAC-6786 Doc review false positive on words/phrases that contain escape characters

### PR DESCRIPTION
**Pull request by kobymeir**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-6786

## Description
[CIAC-6786](https://jira-hq.paloaltonetworks.local/browse/CIAC-6786)
Doc review false positive on words/phrases that contain escape characters

### CODE REVIEW

File: demisto_sdk/commands/doc_reviewer/doc_reviewer.py [hunk 1]
```diff
@@ -408,10 +406,22 @@ def suggest_if_misspelled(self, word: str) -> Optional[Set]:
             return candidates
         return None
 
+
+    def check_sentence(self, sentence):
```
Comment (dorschw) [inline demisto_sdk/commands/doc_reviewer/doc_reviewer.py:410]: add type hints

Comment (dorschw) [inline demisto_sdk/commands/doc_reviewer/doc_reviewer.py:410]:
```suggestion
    def check_sentence(self, sentence: str):
```

File: demisto_sdk/commands/doc_reviewer/doc_reviewer.py [hunk 2]
```diff
@@ -34,6 +34,9 @@
 from demisto_sdk.commands.doc_reviewer.known_words import KNOWN_WORDS
 from demisto_sdk.commands.doc_reviewer.rn_checker import ReleaseNotesChecker
 
+def remove_escape_characters(sentence: str) -> str:
+    return sentence.replace('\\n', ' ').replace('\\r', ' ').replace('\\b', ' ').replace('\\f', ' ').replace('\\t', ' ')
+
```
Comment (bziser) [inline demisto_sdk/commands/doc_reviewer/doc_reviewer.py:39]:
```suggestion
def remove_escape_characters(sentence: str) -> str:
    escape_chars = ['\\n', '\\r', '\\b', '\\f', '\\t']
    for escape_char in escape_chars:
        sentence = sentence.replace(escape_char, ' ')
    return sentence
```

WDYT @kobymeir ? Isn't it more clear?

**Review by bziser — COMMENTED** [review]
@kobymeir Please add a test that shows the difference between the previous and current state :)

File: demisto_sdk/commands/doc_reviewer/doc_reviewer.py [hunk 3]
```diff
@@ -169,14 +175,13 @@ def is_camel_case(self, word):
...
         """split camel case word into sub-words"""
-        tokens = re.compile("([A-Z]?[a-z]+)").findall(camel)
-        for token in tokens:
-            # double space to handle capital words like IP/URL/DNS that not included in the regex
-            camel = camel.replace(token, f" {token} ")
-
-        return camel.split()
+        # Use regular expressions to split the CamelCase word into individual words
+        matches = re.finditer(
+            ".+?(?:(?<=[a-z])(?=[A-Z])|(?<=[A-Z])(?=[A-Z][a-z])|$)", camel
```
Comment (tkatzir) [inline demisto_sdk/commands/doc_reviewer/doc_reviewer.py:182]: Can the re be compiled before, and only used here?

File: demisto_sdk/commands/doc_reviewer/doc_reviewer.py [hunk 4]
```diff
@@ -35,6 +35,13 @@
 from demisto_sdk.commands.doc_reviewer.rn_checker import ReleaseNotesChecker
 
 
+def remove_escape_characters(sentence: str) -> str:
```
Comment (tkatzir) [inline demisto_sdk/commands/doc_reviewer/doc_reviewer.py:38]: Needs a unit test.

File: demisto_sdk/commands/doc_reviewer/doc_reviewer.py [hunk 5]
```diff
@@ -169,14 +175,13 @@ def is_camel_case(self, word):
         return False
 
     @staticmethod
-    def camel_case_split(camel):
+    def camel_case_split(camel: str):
```
Comment (tkatzir) [inline demisto_sdk/commands/doc_reviewer/doc_reviewer.py:178]: Needs a unit test.

File: demisto_sdk/commands/doc_reviewer/doc_reviewer.py [hunk 6]
```diff
@@ -408,10 +413,21 @@ def suggest_if_misspelled(self, word: str) -> Optional[Set]:
             return candidates
         return None
 
+    def check_sentence(self, sentence: str):
```
Comment (tkatzir) [inline demisto_sdk/commands/doc_reviewer/doc_reviewer.py:416]: Needs a unit test.

**Review by tkatzir — COMMENTED** [review]
Very good!
Some comments.

File: demisto_sdk/commands/doc_reviewer/doc_reviewer.py (hunk 5 above)
Comment (kobymeir) [reply demisto_sdk/commands/doc_reviewer/doc_reviewer.py:178]: there is one

File: demisto_sdk/commands/doc_reviewer/doc_reviewer.py (hunk 6 above)
Comment (kobymeir) [reply demisto_sdk/commands/doc_reviewer/doc_reviewer.py:415]: covered by integrations test

File: demisto_sdk/commands/doc_reviewer/doc_reviewer.py [hunk 7]
```diff
@@ -171,12 +175,8 @@ def is_camel_case(self, word):
     @staticmethod
     def camel_case_split(camel):
```
[the remark this answers is not in the document]
Comment (kobymeir) [reply demisto_sdk/commands/doc_reviewer/doc_reviewer.py]: tried, too complex due to mypy issues

File: demisto_sdk/commands/doc_reviewer/doc_reviewer.py (hunk 4 above)
Comment (kobymeir) [reply demisto_sdk/commands/doc_reviewer/doc_reviewer.py]: added

**Review by dantavori — APPROVED** [review]

**Pull request merged**